### ISSUE TITLE
migration from 1.2.32 to 1.2.33 breaks app

### ISSUE THREAD

**Issue by cescoferraro** [gatsbyjs/gatsby#26446]
## Summary

yarn add --exact gatsby-plugin-mdx@1.2.32 works fine
yarn add --exact gatsby-plugin-mdx@1.2.33/34 breaks

## Relevant information

https://github.com/agenciaglobal/web

```
warn Module build failed (from ./node_modules/gatsby-plugin-mdx/loaders/mdx-loader.js):
Error: [BABEL] unknown: Preset /* your preset */ requires a filename to be set when babel is called directly,

babel.transform(code, { filename: 'file.ts', presets: [/* your preset */] });

See https://babeljs.io/docs/en/options#filename for more information.
    at validateIfOptionNeedsFilename (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/@babel/core/lib/config/full.js:274:11)
    at /Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/@babel/core/lib/config/full.js:286:52
    at Array.forEach (<anonymous>)
    at validatePreset (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/@babel/core/lib/config/full.js:286:25)
    at loadPresetDescriptor (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/@babel/core/lib/config/full.js:293:3)
    at loadPresetDescriptor.next (<anonymous>)
    at recurseDescriptors (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/@babel/core/lib/config/full.js:107:30)
    at recurseDescriptors.next (<anonymous>)
    at loadFullConfig (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/@babel/core/lib/config/full.js:142:6)
    at loadFullConfig.next (<anonymous>)
    at Function.transform (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/@babel/core/lib/transform.js:25:45)
    at transform.next (<anonymous>)
    at evaluateSync (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/gensync/index.js:244:28)
    at Function.sync (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/gensync/index.js:84:14)
    at Object.transform (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/@babel/core/lib/transform.js:36:54)
    at module.exports (/Users/cescoferraro/go/src/github.com/agenciaglobal/web/node_modules/gatsby-plugin-mdx/utils/extract-exports.js:9:9)`
<!-- Provide as much useful information as you can -->
```

### File contents (if changed)

https://github.com/agenciaglobal/web/blob/master/babel.config.js
`gatsby-config.js`: N/A 
https://github.com/agenciaglobal/web/blob/master/gatsby-config.js
`package.json`: N/A 
https://github.com/agenciaglobal/web/blob/master/package.json
`gatsby-node.js`: N/A 
https://github.com/agenciaglobal/web/blob/master/gatsby-node.js
`gatsby-browser.js`: N/A 
https://github.com/agenciaglobal/web/blob/master/gatsby-browser.js
`gatsby-ssr.js`: N/A

**Comment by vladar** [comment]
@pvdz Can something like this be caused by recent MDX updates?

**Comment by pvdz** [comment]
Anything is possible :) I check into this next week.

**Comment by pvdz** [comment]
Had to drop the yarn.lock and some other things to reproduce your error.

The problem appears to be caused by a "new" requirement by Babel, which requires a `filename` property (as per the error).

I'll merge and publish a fix soon. Meanwhile you can unblock yourself by going into `node_modules/gatsby-plugin-mdx/utils/extract-exports.js` and adding a `filename` field with any value to the call to `babel.transform`.

Note that afterwards your build produces a bunch of different errors of this kind:
```
./node_modules/browserslist/node.js 161:26-33
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
```

But I believe this is unrelated to mdx. If this is turns out to be a bug in Gatsby please open a new issue for it.

Thank you for reporting the issue :purple_heart:
